wai-middleware-prometheus/src/Network/Wai/Middleware/Prometheus.hs

```diff
@@ -54,22 +55,35 @@ requestLatency = Prom.unsafeRegister $ Prom.vector ("handler", "method", "status
     where info = Prom.Info "http_request_duration_seconds"
                            "The HTTP request latencies in seconds."
 
+-- | Instrument a WAI app with a user-defined labelling function.
+--
+-- If you use this function you will likely want to override the default value
+-- of 'prometheusInstrumentApp' to be false so that your app does not get double
+-- instrumented.
+instrumentWithLabels ::
+     (Wai.Request -> Text) -- ^ The labelling function used to identify this app
```

Comment:
It's not clear here how the `Text` return value will be used. Perhaps say that this will be the value of the `"handler"` label?

I think you might want to carry this idea through to the function's doc comment and maybe even its name. It's not really a general purpose "user-defined labelling function", it's just a function for setting "handler".

Reply:
Agreed, good point. I updated the names.